_rules/inline-link-in-paragraph-is-distinguishable-be4d0c.md

```diff
@@ -0,0 +1,535 @@
+</p>
+```
+
+#### Passed Example 16
+
+This link had a descendant element that is distinguishable by color (hue) from the other text in the same line and has a distinguishable bottom border when it is hovered or focused.
+
+```html
+<style>
+	a {
+		text-decoration: none;
+	}
+
+	a:hover span,
+	a:focus span {
+		border-bottom: 2px solid currentColor;
+	}
+</style>
+
+<p>
+	Read about WAI on the <a href="http://w3.org/WAI"><span>WAI webpage</span></a
+	>.
+</p>
+```
+
+### Failed
+
+#### Failed Example 1
+
+This link, that is distinguishable by color (hue) from the other text in the same line, has no other visual cues of being recognized as a link with the underline removed. Even though the link has a different color (blue) from the other text (black) the low difference in contrast between them means they are not distinguishing colors.
+
+```html
+<style>
+	p {
+		color: black;
+		background-color: white;
+	}
+	a.test {
+		color: blue;
+		background-color: white;
+		text-decoration: none;
+	}
+</style>
+<p>Read about WAI on the <a class="test" href="http://w3.org/WAI">WAI webpage</a>.</p>
+```
+
+#### Failed Example 2
+
+This link, that is distinguishable by color (hue) from the other text in the same line, has a different style because of its color and foreground color. But when the link has been visited, the style is the same of the other text on the same line.
+
+```html
+<style>
+	p {
+		color: black;
+		background-color: white;
+	}
+	a {
+		color: lightblue;
+		background-color: #111111;
+		text-decoration: none;
+	}
+	a:visited {
+		color: black;
+		background-color: white;
+	}
+</style>
+<p>Read about WAI on the <a href="https://www.w3.org/WAI/">WAI webpage</a>.</p>
+```
+
+### Inapplicable
+
+#### Inapplicable Example 1
+
+There is no [semantic link][] element.
+
+```html
+<p>Read about WAI on the <u>underlined text</u>.</p>
+```
+
+#### Inapplicable Example 2
+
+This link is not [visible][].
+
+```html
+<p>Read about WAI on the <a href="http://w3.org/WAI" style="visibility:hidden">WAI webpage</a>.</p>
+```
+
+#### Inapplicable Example 3
+
+This link is the only text rendered in its line.
+
+```html
+<p><a href="http://w3.org/WAI">WAI webpage</a></p>
+```
+
+#### Inapplicable Example 4
+
+Each link is the only text rendered in their respective lines.
+
+```html
+<ul>
+	<li><a href="https://www.w3.org/WAI/WCAG21/Understanding/non-text-content.html">SC 1.1.1</a></li>
+	<li><a href="https://www.w3.org/WAI/WCAG21/Understanding/audio-only-and-video-only-prerecorded">SC 1.2.1</a></li>
+</ul>
+```
+
+#### Inapplicable Example 5
+
+There is no text belonging to non [semantic links][semantic link] in this line.
+
+```html
+<p><a href="https://www.w3.org">W3C </a><span role="link" onclick="location='https://www.w3.org/WAI/'">WAI</span></p>
+```
+
+#### Inapplicable Example 6
+
+This link is not distinguishable by color from the other text rendered in its line.
```

Comment:
Need to explain that the hue is the same 😉 because the hex codes are different, which makes it very confusing that they are not distinguishable by color.

Actually, it might be even clearer if the colors are written in HSL syntax directly…

Reply:
Good suggestion. I've added the HSL value as a comment.